src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,264 @@
+        // Most EVM chain IDs are well below this limit, but we check to be safe
+        if (_lifiChainId > type(uint64).max) {
+            revert InvalidConfig();
+        }
+
+        return uint64(_lifiChainId);
+    }
+
+    function _buildReward(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData,
+        bool isNative,
+        uint256 totalAmount
+    ) private view returns (IEcoPortal.Reward memory) {
+        IEcoPortal.TokenAmount[] memory rewardTokens;
+        if (!isNative) {
+            rewardTokens = new IEcoPortal.TokenAmount[](1);
+            rewardTokens[0] = IEcoPortal.TokenAmount({
+                token: _bridgeData.sendingAssetId,
+                amount: totalAmount
+            });
+        } else {
+            rewardTokens = new IEcoPortal.TokenAmount[](0);
+        }
+
+        return
+            IEcoPortal.Reward({
+                creator: msg.sender,
+                prover: _ecoData.prover,
+                deadline: _ecoData.rewardDeadline,
+                nativeAmount: isNative ? totalAmount : 0,
+                tokens: rewardTokens
+            });
+    }
+
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) internal {
+        // Calculate values once
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
+        uint256 totalAmount = _bridgeData.minAmount + _ecoData.solverReward;
+
+        // Build reward for the intent
+        IEcoPortal.Reward memory reward = _buildReward(
+            _bridgeData,
+            _ecoData,
+            isNative,
+            totalAmount
+        );
+
+        // Get the destination chain ID in Eco format
+        uint64 destination = _getEcoChainId(_bridgeData.destinationChainId);
+
+        // Prepare token approval if needed for ERC20 tokens
+        if (!isNative) {
+            LibAsset.maxApproveERC20(
+                IERC20(_bridgeData.sendingAssetId),
+                address(PORTAL),
+                totalAmount
+            );
+        }
+
+        // Publish and fund the intent with encoded route
+        PORTAL.publishAndFund{ value: isNative ? totalAmount : 0 }(
+            destination,
+            _ecoData.encodedRoute,
+            reward,
+            false // allowPartial
+        );
+
+        _emitEvents(_bridgeData, _ecoData);
+    }
+
+    function _validateEcoData(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) private pure {
+        // encodedRoute is required for all chains
+        if (_ecoData.encodedRoute.length == 0) {
+            revert InvalidConfig();
+        }
+
+        // Validation for NON_EVM_ADDRESS receiver
+        if (
+            _bridgeData.receiver == NON_EVM_ADDRESS &&
+            _ecoData.nonEVMReceiver.length == 0
+        ) {
+            revert InvalidReceiver();
+        }
+
+        // For standard receivers, check address match
```

Comment:
if the facet does not support destination calls, we can simplify this logic:
```!_bridgeData.hasDestinationCall && ``` >> remove

Do we have any way to extract the receiver from the encodedRoute?

Reply:
fixed: 19147bbfa8d02cae69c23960e3241ad05c606b7b